x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/container/QueryContainer.java

```diff
@@ -182,7 +182,9 @@ public BitSet columnMask(List<Attribute> columns) {
                     .innerId() : alias.id()) : null;
             for (int i = 0; i < fields.size(); i++) {
                 Tuple<FieldExtraction, ExpressionId> tuple = fields.get(i);
-                if (tuple.v2().equals(id) || (aliasId != null && tuple.v2().equals(aliasId))) {
+                // if the index is already set there is a collision,
+                // so continue searching for the other tuple with the same id
+                if ((tuple.v2().equals(id) || (aliasId != null && tuple.v2().equals(aliasId))) && !mask.get(i)) {
```

Comment:
I would make this condition a bit differently: change the order of the conditions and use explicit `=false` instead of `!`. Something like this: `(mask.get(i) == false && (tuple.v2().equals(id) || (aliasId != null && tuple.v2().equals(aliasId)))`